openff/toolkit/utils/rdkit_wrapper.py

```diff
@@ -1580,6 +1580,10 @@ def from_smiles(
         elif hydrogens_are_explicit:
             for atom_idx in range(rdmol.GetNumAtoms()):
                 atom = rdmol.GetAtomWithIdx(atom_idx)
+                # Issue #1697: GetTotalNumHs() counts the number of hydrogens that are
+                # not present in the graph, so explicit hydrogens are not counted.
```

Comment:
(blocking) move this comment down to just above the line that uses `GetTotalNumHs`. Also expand on why the name is so counterintuitive compared to its behavior (you'd mentioned that there's a named argument that defaults to True, that's worth expanding on)